samtranslator/swagger/swagger.py

```diff
@@ -824,16 +823,32 @@ def add_resource_policy(self, resource_policy, path, api_id, stage):
 
         if source_vpc_whitelist is not None:
             resource_list = self._get_method_path_uri_list(path, api_id, stage)
-            for endpoint in source_vpc_whitelist:
-                self._add_vpc_resource_policy_for_method(endpoint, "StringNotEquals", resource_list)
+            self._add_vpc_resource_policy_for_method(source_vpc_whitelist, "StringNotEquals", resource_list)
 
         if source_vpc_blacklist is not None:
             resource_list = self._get_method_path_uri_list(path, api_id, stage)
-            for endpoint in source_vpc_blacklist:
-                self._add_vpc_resource_policy_for_method(endpoint, "StringEquals", resource_list)
+            self._add_vpc_resource_policy_for_method(source_vpc_blacklist, "StringEquals", resource_list)
 
+        self._doc[self._X_APIGW_POLICY] = self.resource_policy
+
+    def add_custom_statements(self, custom_statements):
         if custom_statements is not None:
-            self._add_custom_statement(custom_statements)
+            if is_intrinsic_if(custom_statements):
+                if_list = custom_statements.get("Fn::If")
+                ## copy the statement created so far
+                original = self.resource_policy.copy()
+                if_true = if_list[1]
+                if_false = if_list[2]
+                if not is_intrinsic_no_value(if_list[1]):
+                    self._add_custom_statement(if_list[1])
+                    if_true = self.resource_policy
+                self.resource_policy = original
+                if not is_intrinsic_no_value(if_list[2]):
+                    self._add_custom_statement(if_list[2])
+                    if_false = self.resource_policy
+                self.resource_policy = make_conditional(if_list[0], if_true, if_false)
+            else:
+                self._add_custom_statement(custom_statements)
```

Comment:
Why do we need to do this? Does it not work to just pass it straight through and let CFN resolve the if statement?

Reply:
The merge between global custom statements and local with `if` won't be handled.

Reply:
Could you add a test case then that tests this flow? I don't see a test case that has both local and global custom statements.